platform/state/platform_state.proto

```diff
@@ -0,0 +1,391 @@
+     * The first round to handle transactions is defined as round `1`.
+     */
+    uint64 round = 3;
+
+    /**
+     * A running event hash.<br/>
+     * This is computed by the consensus event stream.
+     * <p>
+     * This will be _deprecated_ once the consensus event stream is retired.
+     */
+    bytes legacy_running_event_hash = 4;
+
+    /**
+     * A consensus timestamp for this state.
+     * <p>
+     * This SHALL be the consensus timestamp of the first transaction in
+     * the current round.
+     */
+    proto.Timestamp consensus_timestamp = 5;
+
+    /**
+     * A version describing the current version of application software.
+     * <p>
+     * This SHALL be the software version that created this state.
+     */
+    SoftwareVersion creation_software_version = 6;
+
+    /**
+     * A SHA-384 hash that is the "epoch" hash of this state.
+     * <p>
+     * This SHALL be updated whenever emergency recovery is performed.
+     */
+    bytes epoch_hash = 7;
+
+    /**
+     * A SHA-384 hash that will be the next "epoch" hash.
+     * <p>
+     * This SHALL be used to update the epoch hash at the next round boundary.<br/>
+     * This SHALL NOT be part of the hash and SHALL NOT be serialized.
+     */
+    bytes next_epoch_hash = 8;
+
+    /**
+     * A number of non-ancient rounds.
+     * <p>
+     * This SHALL be the count of rounds considered non-ancient.
+     */
+    uint32 rounds_non_ancient = 9;
+
+    /**
+     * A snapshot of the consensus state at the end of the round.
+     * <p>
+     * This SHALL be used for restart/reconnect.
+     */
+    ConsensusSnapshot consensus_snapshot = 10;
+
+    /**
+     * A timestamp for the next scheduled time when a freeze will start.
+     * <p>
+     * If a freeze is not scheduled, this SHALL NOT be set.
+     */
+    proto.Timestamp freeze_time = 11;
+
+    /**
+     * A timestamp for the last time a freeze was performed.<br/>
+     * If not set, there has never been a freeze.
+     */
+    proto.Timestamp last_frozen_time = 12;
+
+    /**
+     * A consensus node software version.<br/>
+     * The software version that enabled birth round mode.
+     * <p>
+     * This SHALL be unset if birth round migration has not yet happened.<br/>
+     * Otherwise, this SHALL be the _first_ software version that enabled birth
+     * round mode.
+     */
+    SoftwareVersion first_version_in_birth_round_mode = 13;
+
+    /**
+     * A consensus round.<br/>
+     * The last round before the birth round mode was enabled.
+     * <p>
+     * This SHALL be -1 if birth round mode has not yet been enabled.
+     */
+    int64 last_round_before_birth_round_mode = 14;
+
+    /**
+     * A consensus generation.<br/>
+     * The lowest judge generation before birth round mode was enabled.
+     * <p>
+     * This SHALL be -1 if birth round mode has not yet been enabled.
+     */
+    int64 lowest_judge_generation_before_birth_round_mode = 15;
```

Comment:
Ideally these would have high index values, since we plan on deprecating these fields once the birth round migration is completed.

Reply:
Done